src/Concordium/Client/Runner.hs

```diff
@@ -1959,14 +2014,18 @@ processContractCmd action baseCfgDir verbose backend =
         Just (InvokerAccount nameOrAddr) -> Just . Types.AddressAccount . naAddr <$> getAccountAddressArg (bcAccountNameMap baseCfg) nameOrAddr
         Just InvokerContract{..} -> Just . Types.AddressContract . ncaAddr <$> getNamedContractAddress (bcContractNameMap baseCfg) icIndexOrName icSubindex
 
-      contrInfo <- withClient backend . withBestBlockHash block $ getContractInfo namedContrAddr
+      (bbHash, contrInfo) <- withClient backend $ do
+        bhInput <- readBlockHashOrDefault Best block
+        bHash <- extractResponseValueOrDie Queries.biBlockHash =<< getBlockInfoV2 bhInput
```

Comment:
So you are doing this a lot, but it is not necessary. It is not wrong, but each request returns in its header the block hash that was used, so if you supply Best you will get back a header blockHash with the actual hash that was used.

this would save these requests. I don't want you to spend time changing things now, but wanted to bring it up.

Reply:
Neat. I'll change it.

Reply:
As I wrote, if this is going to delay merging this PR then do not do it.

Reply:
I don't think it will, but if it seems to be cumbersome I'll leave it as is.

Reply:
I gave it a shot, but it took too long, so I opted not to.